kiwi/system/result.py

```diff
@@ -162,7 +160,7 @@ def dump(self, filename: str, portable: bool = True) -> None:
             )
 
     @staticmethod
-    def load(filename: str) -> result_type:
```

Comment:
I get `error: A function returning TypeVar should receive at least one argument containing the same Typevar` 
This issue AFAIU comes from [this check](https://github.com/python/mypy/pull/13166), hope that clarifies the error message
IMO, without debugging more, a possible solution could be to ignore it, such as
```python
def load(filename: str) -> result_type  # type: ignore[type-var]
```

Reply:
That I tried which results in

```
check run-test: commands[0] | flake8 --statistics -j auto --count /home/ms/Project/kiwi/kiwi
kiwi/system/result.py:165:32: F821 undefined name 'Result'
1     F821 undefined name 'Result'
1
```

which was the reason for using

```
result_type = TypeVar('result_type', bound='Result')
```

Reply:
Yes, but that is fixable adding
```python
from __future__ import annotations
```
at the beggining of the file ([more info](https://docs.python.org/3/whatsnew/3.7.html#pep-563-postponed-evaluation-of-annotations) notice _It will become the default in Python 3.10._ in that doc)
or 
```python
def load(filename: str) -> Result:  # noqa: F821
```
Now, it is a decision of 
- do we want to keep `Result` (type method returns) ? My answer is yes
- do we want to add  `#noqa` ? My opinion is no, because that would hide possible errors but no strong opinion
- do we want to remove the type ? I think we should keep it

my 2 cents

Reply:
we could just make SUSE build this against a newer Python in SLE 15. They have Python 3.6, 3.8, and 3.10 now.